### ISSUE TITLE
PCB Schematic?

### ISSUE THREAD

**Issue by mtfurlan** [rbenamotz/LEMPA#3]
I can't seem to find the actual schematic for the PCB?

I've been working on a kinda similar thing, designed around a USB<-->UART FT232 instead of a pi, and I'm curious how you're dealing with GPIO0 and 3V3 on the ESP to rest it properly and such things.
I've been planning to use the reset circuit off the D1 mini, but I'm curious as to how you approached it.

**Comment by rbenamotz** [comment]
Hi

You can find the schematics here:

https://easyeda.com/roey/flower-tzarvan_copy


I am taking a very simple approach toward ESP (but it works): One of the pins is for GPIO0 and it's always grounded so once you connect the ESP, it's on program mode. Once programming is done you unplug the ESP.
On future version I am adding option for reset.

Another issue I approacheed with ESP was the fact that in many boards I have ATMega32 + ESP on the same board connected via serial. When that happens I cannot program the ESP as the ATMega keeps the TX pin HIGH all the time. The solution I used is the following:
1. I created a very small ATMega firmware which just disables the TX `(_SFR_BYTE(UCSRB) &= ~_BV(TXEN));`
2. Added a script capbility to the programmer (composite in the JSON)
3. The programmer, on single click, does the follwing: 1. Load the dummy firmware to ATMega 2. Loads the ESP (as its serial is free now) 3. Loads the real firmware to the ATMega


Hope this helps


Roey

**Comment by mtfurlan** [comment]
Oh, that makes sense.
Thank you!
